PennMobile/src/main/java/com/pennapps/labs/pennmobile/LaundryFragment.java

```diff
@@ -0,0 +1,216 @@
+                if (hall_no != -1) {
+                    args.putInt(getString(R.string.laundry_hall_no), hall_no);
+                    getArguments().putInt(getString(R.string.laundry_hall_no), -1);
+                }
+            }
+            fragment.setArguments(args);
+            fragmentManager.beginTransaction()
+                    .replace(R.id.laundry_fragment, fragment)
+                    .addToBackStack(null)
+                    .commit();
+        } else if (lh.getIds().size() == 1) {
+            Fragment fragment = new LaundryMachineFragment();
+            args.putParcelable(getString(R.string.laundry), lh.getIds().get(0));
+            fragment.setArguments(args);
+            fragmentManager.beginTransaction()
+                    .replace(R.id.laundry_fragment, fragment)
+                    .addToBackStack(null)
+                    .commit();
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        getActivity().setTitle(R.string.laundry);
+        mActivity.setNav(R.id.nav_laundry);
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        ButterKnife.unbind(this);
+    }
+
+    public static void setSummary(int avail, int used, int idoffset, RelativeLayout rl, Context context) {
+        if (idoffset == 0) {
+            idoffset = 0;
+        }
+        LinkedList<ImageView> vertical = new LinkedList<>();
+        ImageView prev = null;
+        int max_col = ROW_CAP;
+        if (avail + used > ROW_CAP) {
+            max_col = (avail + used) / 2;
+        }
+        for (int i = 0; i < avail + used; i++) {
+            RelativeLayout.LayoutParams layparam = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT,
+                    RelativeLayout.LayoutParams.WRAP_CONTENT);
+            if (i < max_col) {
+                layparam.addRule(RelativeLayout.ALIGN_PARENT_TOP);
+            } else if (i == max_col) {
+                layparam.addRule(RelativeLayout.BELOW, vertical.getLast().getId());
+            } else {
+                layparam.addRule(RelativeLayout.BELOW, vertical
+                        .get(vertical.size() - 2).getId());
+            }
+            if (i % max_col == 0) {
+                layparam.addRule(RelativeLayout.ALIGN_PARENT_LEFT);
+            } else if (prev != null) {
+                layparam.addRule(RelativeLayout.RIGHT_OF, prev.getId());
+            }
+            layparam.setMargins(0, 0, 7, 5);
+            ImageView imageView = new ImageView(context);
+            if (i < avail) {
+                imageView.setImageResource(R.drawable.green_circle);
+            } else {
+                imageView.setImageResource(R.drawable.red_circle);
+            }
+            imageView.setId((idoffset + 1) * 300 + i);
+            rl.addView(imageView, layparam);
+            if (i % max_col == 0) {
```

Comment:
I'm having a really hard time deciphering what all this does. Is it possible to combine this check with the check on 196? Also, can this be reorganized to make it easier to read?